exercises/sending-signals-external/practice/workflow.py

```diff
@@ -0,0 +1,90 @@
+import asyncio
+from datetime import timedelta
+
+from temporalio import workflow
+from temporalio.exceptions import ApplicationError
+
+# Import activity, passing it through the sandbox without reloading the module
+with workflow.unsafe.imports_passed_through():
+    from activities import PizzaOrderActivities
+    from shared import Bill, OrderConfirmation, PizzaOrder
+
+@workflow.defn
+class PizzaOrderWorkflow:
+    def __init__(self) -> None:
+        self._pending_confirmation: asyncio.Queue[str] = asyncio.Queue()
+
+    @workflow.run
+    async def order_pizza(self, order: PizzaOrder) -> OrderConfirmation:
+        workflow.logger.info(f"order_pizza workflow invoked with {input}")
```

Comment:
where is input defined?

Reply:
Good catch — I had actually borrowed this part directly from https://github.com/temporalio/edu-102-python-code/blob/main/exercises/debug-activity/solution/workflow.py and didn't look at it too closely because I was focused on getting the Signals stuff working, but it looks like it's missing there too, and this doesn't throw an error in the logs so I don't think Mason or I ever spotted that it's undefined:

```
INFO:temporalio.workflow:order_pizza workflow invoked with <temporalio.worker.workflow_sandbox._importer._ThreadLocalCallable object at 0x10d474790> ({'attempt': 1, 'namespace': 'default', 'run_id': 'cdc01f97-ba96-4548-a83d-a1ee8bee1ed5', 'task_queue': 'pizza-tasks', 'workflow_id': 'pizza-workflow-order-XD001', 'workflow_type': 'PizzaOrderWorkflow'})
```

I'm tempted to just change this to "order_pizza workflow invoked" so the logs look like this:

```
INFO:temporalio.workflow:order_pizza workflow invoked ({'attempt': 1, 'namespace': 'default', 'run_id': '5d3a76f1-e864-4f68-a923-9e50e6a8be9c', 'task_queue': 'pizza-tasks', 'workflow_id': 'pizza-workflow-order-XD001', 'workflow_type': 'PizzaOrderWorkflow'})
```

Since I assume the original purpose of using `{input}` here has been forgotten.

Reply:
Whoops. Copy past that no one ever caught. Surprised that worked.